### PR TITLE
[WIP] Add `HashStableEq` trait, and use it in query code

### PR DESCRIPTION
**Pull request by Aaron1011**

### CODE REVIEW

**Comment by Kobzol** [comment]
By the way, the `ToStableHashKey` is now probably no longer needed, since the items in collections without a defined order are no longer sorted when performing stable hashing. So if `HashStable` is getting refactored/changed a lot, it might be worth it to also remove this trait at the same time.

File: compiler/rustc_middle/src/ty/sty.rs [hunk 1]
```diff
@@ -1483,22 +1483,35 @@ pub enum RegionKind {
     ReErased,
 }
 
-#[derive(Copy, Clone, PartialEq, Eq, Hash, TyEncodable, TyDecodable, Debug, PartialOrd, Ord)]
+#[derive(
```
Comment (Aaron1011) [inline compiler/rustc_middle/src/ty/sty.rs:1486]: This formatting is really obnoxious - is there a way to configure rustfmt to full up each line with arguments before moving to the next line?

**Comment by JohnCSimon** [comment]
@Aaron1011 
Ping from triage: I'm closing this due to inactivity, Please reopen when you are ready to continue with this. 
Note: if you do please open the PR **BEFORE** you push to it, else you won't be able to reopen - this is a quirk of github. 
Thanks for your contribution.

@rustbot label: +S-inactive

**Pull request closed without merging**